src/vs/workbench/api/common/extHost.api.impl.ts

```diff
@@ -309,6 +309,9 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 			get shell() {
 				return extHostTerminalService.getDefaultShell(false);
 			},
+			get onDidChangeShell() {
+				return extHostTerminalService.onDidChangeShell;
```

Comment:
Missing proposed API check

Reply:
Added `envShellEvent` to API proposals type, and called the checker method as above.